packages/core/src/services/workspaces/usage.ts

```diff
@@ -0,0 +1,52 @@
+import { count, gte } from 'drizzle-orm'
+
+import { MAX_FREE_RUNS, Workspace, WorkspaceUsage } from '../../browser'
+import { database } from '../../client'
+import { PromisedResult, Result } from '../../lib'
+import { EvaluationResultsRepository } from '../../repositories'
+import { DocumentLogsRepository } from '../../repositories/documentLogsRepository'
+
+export async function computeWorkspaceUsage(
+  workspace: Workspace,
+  db = database,
+): PromisedResult<WorkspaceUsage> {
+  const { scope: documentLogsScope } = new DocumentLogsRepository(
+    workspace.id,
+    db,
+  )
+  const { scope: evaluationResultsScope } = new EvaluationResultsRepository(
+    workspace.id,
+    db,
+  )
+
+  const createdAtDate = workspace.createdAt
+  const currentDate = new Date(Date.now())
+
+  const day = createdAtDate.getDate()
+  const month =
+    (currentDate.getMonth() - (currentDate.getDate() < day ? 1 : 0)) % 12
+  const year = currentDate.getFullYear()
+
+  const latestRenewalDate = new Date(year, month, day)
```

Comment:
Free plan is renewed in a monthly basis on the same day of creation each month (not every 30 days). This is how I calculate which day was it last renewed based on the creation date.